Shared/Samples/Create KML multi-track/CreateKMLMultiTrackView.swift

```diff
@@ -0,0 +1,185 @@
+                        .disabled(asyncAction != nil && asyncAction != .recordTrack)
+                        
+                        Spacer()
+                        
+                        Button("Save", systemImage: "square.and.arrow.down") {
+                            asyncAction = .saveKMLMultiTrack
+                        }
+                        .disabled(asyncAction != nil || model.tracks.isEmpty)
+                    }
+                }
+            }
+            .task(id: asyncAction) {
+                // Runs the asynchronous action.
+                guard let asyncAction else {
+                    return
+                }
+                defer { self.asyncAction = nil }
+                
+                do {
+                    switch asyncAction {
+                    case .recordTrack:
+                        for await location in model.locationDisplay.$location where location != nil {
+                            model.addTrackElement(atPoint: location!.position)
+                            statusText = "Recording KML track. Elements added: \(model.trackElements.count)"
+                        }
+                        
+                        statusText = "Tap record to capture KML track elements."
+                    case .saveKMLMultiTrack:
+                        await model.locationDisplay.dataSource.stop()
+                        
+                        try await model.saveKMLMultiTrack()
+                        multiTrack = try await model.loadKMLMultiTrack()
+                        
+                        statusText = "Saved KML multi-track to 'HikingTracks.kmz'."
+                    case .reset:
+                        model.reset()
+                        multiTrack = nil
+                        await mapViewProxy.setViewpointScale(model.locationDisplay.initialZoomScale)
+                        
+                        fallthrough
+                    case .startNavigation:
+                        try await model.startNavigation()
+                        statusText = "Tap record to capture KML track elements."
+                    }
+                } catch {
+                    self.error = error
+                }
+            }
+            .task {
+                // Monitors the auto pan mode to determine if recenter button should be enabled.
+                for await autoPanMode in model.locationDisplay.$autoPanMode {
+                    recenterIsEnabled = autoPanMode != .navigation
+                }
+            }
+            .errorAlert(presentingError: $error)
+        }
+    }
+}
+
+private extension CreateKMLMultiTrackView {
+    /// A picker for selecting a track from a KML multi-track.
+    struct TrackPicker: View {
+        /// The KML tracks options shown in the picker.
+        private let tracks: [KMLTrack]
+        
+        /// The proxy for setting map view's viewpoint to the selected track's geometry.
+        private let mapViewProxy: MapViewProxy
```

Comment:
Giving this view the whole map view proxy seems overkill. It might have been better to have it take a closure that is called when selection changes, then the caller could provide a closure that uses the map view proxy. This wouldn't even need to know about it.